### PR TITLE
Override the Kotlin compilation task inputs

### LINKED ISSUE #65: Usage of absolute path as input
Recently after comparing build scans produced on developer machine and CI node we noticed following:
> Input 'kotlinCompile.com.squareup.anvil.compiler.src-gen-dir' is in both builds but the values are different.

This produces remote build cache misses for following tasks as paths are different:
- `org.jetbrains.kotlin.gradle.internal.KaptGenerateStubsTask`
- `org.jetbrains.kotlin.gradle.internal.KaptWithoutKotlincTask`
- `org.jetbrains.kotlin.gradle.tasks.KotlinCompileWithWorkers`

I took a look in the source code and found this line: https://github.com/square/anvil/blob/9fb3c5f5b8a5134dd97d21d740271dcb276c6e0c/gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilSubplugin.kt#L46 that returns absolute path to the Kotlin plugin. Kotlin plugin itself sets this as task inputs via https://github.com/JetBrains/kotlin/blob/34b55dbeb3a65b39b3585a646b5a0d2b1ce4dd96/libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinPlugin.kt#L1078

I am not sure what proper fix should be. Probably either using `srcGenDir.relativeTo(project.buildDir)` or asking Kotlin team to add relative `kind` type to [FilesSubpluginOptions](https://github.com/JetBrains/kotlin/blob/master/libraries/tools/kotlin-gradle-plugin-api/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinGradleSubplugin.kt#L36).

### PR DESCRIPTION
**Pull request by vRallev**
Override the Kotlin compilation task inputs to avoid build cache misses on different machines.

Fixes #65

### CODE REVIEW

File: gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilSubplugin.kt [hunk 1]
```diff
@@ -39,12 +39,23 @@ class AnvilSubplugin : KotlinGradleSubplugin<AbstractCompile> {
     // for this specific compile task will be included in the task output. The output of different
     // compile tasks shouldn't be mixed.
     val srcGenDir = File(project.buildDir, "anvil/src-gen-${kotlinCompile.name}")
```
Comment (cdeange) [inline gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilSubplugin.kt:41]: nit: use `File.separator` for platform-agnostic file-path building.

Comment (vRallev) [reply gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilSubplugin.kt:41]: You or someone else pointed this out already and then I forgot to fix it. This is a good PR to fix it.

Comment (cdeange) [reply gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilSubplugin.kt:41]: It was me 😄

Comment (vRallev) [reply gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilSubplugin.kt]: I feel bad

**Pull request merged**